### PR TITLE
feat(case-assist): add Case Assist client

### PR DESCRIPTION
**Pull request by lbergeron**
https://coveord.atlassian.net/browse/SFINT-3835

This PR adds a strongly-typed client to log the Case Assist events. See [Log Case Assist Events](https://docs.coveo.com/en/3437/service/log-case-assist-events) for details regarding the information that are passed to the `svc` plugin.

### CODE REVIEW

File: src/caseAssist/caseAssistClient.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        this.svc.setAction('ticket_cancel', {
+            reason: 'Quit',
+        });
+        this.svc.setTicket(meta.ticket);
+        return this.sendClickEvent();
+    }
+
+    public logCaseSolved(meta: CaseSolvedMetadata) {
+        this.svc.setAction('ticket_cancel', {
+            reason: 'Solved',
```
Comment (louis-bompart) [inline src/caseAssist/caseAssistClient.ts:93]: Do you think the reason should be extracted into an `enum`?

File: src/caseAssist/caseAssistClient.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    }
+
+    public logSelectFieldSuggestion(meta: SelectFieldSuggestionMetadata) {
+        this.svc.setAction('ticket_classification_click', meta.suggestion);
+        this.svc.setTicket(meta.ticket);
+        return this.sendClickEvent();
+    }
+
+    public logSelectDocumentSuggestion(meta: SelectDocumentSuggestionMetadata) {
+        this.svc.setAction('suggestion_click', meta.suggestion);
```
Comment (louis-bompart) [inline src/caseAssist/caseAssistClient.ts:63]: Do you think the action (e.g. `suggestion_click` etc) should be in an (or multiple) enums?

**Review by louis-bompart — APPROVED** [review]
lgtm overall 👍

File: src/caseAssist/caseAssistClient.ts (hunk 1 above)
Comment (lbergeron) [reply src/caseAssist/caseAssistClient.ts:93]: Yes I can add an enum for that.

File: src/caseAssist/caseAssistClient.ts (hunk 2 above)
Comment (lbergeron) [reply src/caseAssist/caseAssistClient.ts:63]: In fact, I should reuse the `CaseAssistActions` enum when setting the actions.

**Review by nathanlb — APPROVED** [review]
LGTM!

File: src/caseAssist/caseAssistClient.ts [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    EnterInterfaceMetadata,
+    MoveToNextCaseStepMetadata,
+    RateDocumentSuggestionMetadata,
+    SelectDocumentSuggestionMetadata,
+    SelectFieldSuggestionMetadata,
+    UpdateCaseFieldMetadata,
+} from './caseAssistActions';
+
+export interface CaseAssistClientOptions extends ClientOptions {
+    enableAnalytics: boolean;
```
Comment (francoislg) [inline src/caseAssist/caseAssistClient.ts:20]: I guess this can be optional ?

**Review by francoislg — APPROVED** [review]
I didn't quite validate the format and everything, but the code itself fits well into what we have and it looks good :+1:

File: src/caseAssist/caseAssistClient.ts (hunk 3 above)
Comment (lbergeron) [reply src/caseAssist/caseAssistClient.ts:20]: I defined the option the same way as in `SearchPageClientOptions` to be consistent. But you are right that this flag could be optional, and should default to `true`.

Comment (lbergeron) [reply src/caseAssist/caseAssistClient.ts]: @francoislg I just updated the PR to make `enableAnalytics` optional and default to `true`. Is that OK with you?

**Pull request merged**